PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/Endpoint/controller_class/configuration_controller.cpp

```diff
@@ -0,0 +1,171 @@
+#include "configuration_controller.h"
+
+/**
+ * @brief Construct a new Configuration Controller and initializing some needed pointers to be used inside it 
+ */
+ConfigurationController::ConfigurationController()
+{
+    configuration = ConfigurationSpace::constructConfigurationSpace();
+    capability =  PCIECapability::constructPCIECapability();
+
+    handler = make_shared<ConfigurationRequestHandler>(configuration, capability);
+    tlpConstructor = make_shared<TLPConstructor>();
+
+    cplD = make_shared<CompletionWithData>();
+    cpl = make_shared<CompletionWithoutData>();
+    cplUR = make_shared<CompletionWithUR>();
+}
+
+/**
+ * @brief Getting the Register number which we need to apply the Configuration request on
+ * 
+ * @param tlp -> The TLP coming to be handled
+ * @return int -> The Register Number to apply the Configuration request on
+ */
+int ConfigurationController::getRegisterNumber(TLP * tlp)
+{
+    /* Here I need a pointer to the ConfigNonHeaderBase class and I should put here instead of (nonBase) */
```

Comment:
Okay, I see that this part also depends on something that needs to be finished first

Reply:
It just needed an explicit casting and I solved this.

Reply:
okay cool 👍